schema/deploy/shipping/views.sql

```diff
@@ -2328,6 +2328,196 @@ grant select
+        case
+          when applied_vac_3[1] is not null then 3
+          when applied_vac_2[1] is not null then 2
+          when applied_vac_1[1] is not null then 1
+          when encountered::date < '2020-12-01' or covid_vax = 'no' then 0
+          else null
+        end as calculated_covid_doses,
+        applied_vac_1[1] as calculated_vac_date_1,
+        applied_vac_1[2] as calculated_vac_name_1,
+        applied_vac_2[1] as calculated_vac_date_2,
+        applied_vac_2[2] as calculated_vac_name_2,
+        applied_vac_3[1] as calculated_vac_date_3,
+        applied_vac_3[2] as calculated_vac_name_3,
+        vac_date_1_out_of_range,
+        vac_date_2_out_of_range,
+        vac_date_3_out_of_range,
+        old_vaccination_status
+      from encounters_with_best_available_vac_applied
+    )
+
+    select
+      individual,
+      encounter_id,
+      encountered,
+      pt_entered_covid_vax,
+      pt_entered_covid_doses,
+      calculated_covid_doses,
+      calculated_vac_date_1,
+      calculated_vac_name_1,
+      calculated_vac_date_2,
+      calculated_vac_name_2,
+      calculated_vac_date_3,
+      calculated_vac_name_3,
+      vac_date_1_out_of_range,
+      vac_date_2_out_of_range,
+      vac_date_3_out_of_range,
+      case
+        when encountered < '2020-12-01' then 'na'
+        when pt_entered_covid_vax = 'no' then 'not_vaccinated'
+          when (vac_date_3_out_of_range is null and  (calculated_vac_name_3 in (values('pfizer'), ('moderna')) and date_part('day', encountered::timestamp - calculated_vac_date_3::timestamp) >= 14))
+            or (vac_date_2_out_of_range is null and calculated_vac_name_1 = 'johnson' and calculated_vac_name_2 in (values('pfizer'), ('moderna'), ('johnson')) and date_part('day', encountered::timestamp - calculated_vac_date_2::timestamp) >= 14) then 'boosted'
+          when vac_date_1_out_of_range is null and calculated_vac_name_1 = 'johnson' and date_part('day', encountered::timestamp - calculated_vac_date_1::timestamp) >= 14 then 'fully_vaccinated'
+          when vac_date_2_out_of_range is null and calculated_vac_name_1 in (values('pfizer'), ('moderna')) and calculated_vac_name_2 in (values('pfizer'), ('moderna')) and date_part('day', encountered::timestamp - calculated_vac_date_2::timestamp) >= 14 then 'fully_vaccinated'
+          when vac_date_1_out_of_range is null and calculated_vac_name_1 in (values('pfizer'), ('moderna'), ('johnson')) and date_part('day', encountered::timestamp - calculated_vac_date_1::timestamp) >= 1 then 'partially_vaccinated'
+          when vac_date_1_out_of_range is null and date_part('day', encountered::timestamp - calculated_vac_date_1::timestamp) = 0 then 'not_vaccinated'
+        when coalesce(vac_date_1_out_of_range, vac_date_2_out_of_range, vac_date_3_out_of_range) = 1 then 'invalid'
+          else 'unknown'
+      end as best_available_vac_status
+    from encounters_with_best_available_vac_applied_dose_count
```

Comment:
Documenting conversation because this confused me at first- there was a request from the end users that `best_available_vac_status` be `boosted` if theres a valid `date3` to even if prior doses were invalid dates.